include/range/v3/algorithm/unstable_remove_if.hpp

```diff
@@ -0,0 +1,78 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Andrey Diduh 2019
+//  Copyright Casey Carter 2019
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+#ifndef RANGES_V3_ALGORITHM_UNSTABLE_REMOVE_IF_HPP
+#define RANGES_V3_ALGORITHM_UNSTABLE_REMOVE_IF_HPP
+
+#include <meta/meta.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/algorithm/find_if.hpp>
+#include <range/v3/algorithm/find_if_not.hpp>
+
+/**
+ * unstable_remove have O(1) complexity for each element remove, unlike remove O(n) [for worst case]
+ * Each erased element overwritten (moved in) with last one. Those, unstable_remove loose elements order.
+ */
+namespace ranges
+{
+    inline namespace v3
+    {
+        /// \ingroup group-concepts
+        template<typename I, typename C, typename P = ident>
+        using UnstableRemovableIf = meta::strict_and<
+                BidirectionalIterator<I>,
+                IndirectPredicate<C, projected<I, P>>,
+                Permutable<I>>;
+
+        /// \addtogroup group-algorithms
+        /// @{
+        struct unstable_remove_if_fn
+        {
+            template <typename I, typename C, typename P = ident,
+                    CONCEPT_REQUIRES_(UnstableRemovableIf<I, C, P>())>
+            I operator()(I first, I last, C pred, P proj = {}) const
```

Comment:
All algorithms take iterator/sentinel pairs.

Reply:
This algorithm requires `last` to be "decrementable". 
As far as I understand [Sentinel Concept], Sentinel used as immobile mark , and can not (must not?) be "decremented".

Please, correct me if I wrong.